### PR TITLE
feat(kstepper): new kongponent [beta]

### PR DESCRIPTION
**Pull request by kaiarrowood**
# Summary

## Vue 3

**Did you create a corresponding PR to add this change to the `beta` branch? (required)**

- [ ] **Yes**, here is a link to the PR: `[link to beta branch PR]`
- [ ] **No**, I did not create a PR for the `beta` branch and I have added the `port to beta branch` tag to this PR.
- [ ] **No**, because `[type your reasons]`

If you have questions, tag `@adamdehaven` or `@kaiarrowood`.

---

## PR Checklist

- [ ] Does not introduce dependencies
- [ ] **Functional:** all changes do not break existing APIs and if so, bump major version.
- [ ] **Tests pass:** check the output of yarn test packages/<Kongponent>
- [ ] **Naming:** the files and the method and prop variables use the same naming conventions as other Kongponents
- [ ] **Framework style:** abides by the essential rules in Vue's style guide
- [ ] **Cleanliness:** does not have formatting issues, unused code (e.g., console.logs), or leftover comments
- [ ] **Docs:** includes a technically accurate README, uses JSDOC where appropriate

### CODE REVIEW

File: src/components/KStepper/KStep.vue [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      width: 100%;
+      position: absolute;
+      top: calc(#{var(--KStepSize, 24px)} / 2);
+      left: 50%;
+      z-index: 0;
+      background-color: var(--KStepDividerColorDefault, var(--grey-300));
+    }
+
+    &.completed::after {
+      background-color: var(--KStepDividerColorCompleted, var(--teal-300));
```
Comment (adamdehaven) [inline src/components/KStepper/KStep.vue:112]: Please remove the KStepper CSS variables; we definitely do not want this level of customization (and they don't exist on the `main` branch)

File: src/components/KStepper/KStepState.vue [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  background: var(--KStepBackgroundColor, var(--white));
+  z-index: 1;
+}
+</style>
+
+<style lang="scss">
+.k-step-state {
+  .k-step-state-icon {
+    height: var(--KStepSize, 24px);
+    width: var(--KStepSize, 24px);
```
Comment (adamdehaven) [inline src/components/KStepper/KStepState.vue:41]: Please remove the KStepper CSS variables; we definitely do not want this level of customization (and they don't exist on the `main` branch)

File: docs/components/stepper.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+  --KStepSize: 40px;
+  --KStepDividerColorCompleted: var(--purple-300);
+  --KStepDividerColorDefault: var(--purple-100);
+  --KStepCompletedColor: var(--purple-300);
+  --KStepDefaultColor: var(--purple-100);
+  --KStepErrorColor: var(--steel-400);
+  --KStepPendingColor: var(--yellow-400);
+}
+</style>
+```
```
Comment (adamdehaven) [inline docs/components/stepper.md:157]: The Theming section should not exist; we definitely do not want this level of customization (and they don't exist on the `main` branch)

File: docs/components/stepper.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+.k-stepper-wrapper {
+  --KStepSize: 40px;
+  --KStepDividerColorCompleted: var(--purple-300);
+  --KStepDividerColorDefault: var(--purple-100);
+  --KStepCompletedColor: var(--purple-300);
+  --KStepDefaultColor: var(--purple-100);
+  --KStepErrorColor: var(--steel-400);
+  --KStepPendingColor: var(--yellow-400);
+}
+</style>
```
Comment (adamdehaven) [inline docs/components/stepper.md:197]: This should be removed

**Review by adamdehaven — CHANGES_REQUESTED** [review]

File: docs/components/stepper.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+## Theming
+
+| Variable                      | Purpose                                     |
+| :---------------------        | :-------------------------------            |
+| `KStepSize`                   | Height and width of step state icon         |
+| `KStepBackgroundColor`        | Background color of the step state icon     |
+| `KStepDividerColorCompleted`  | Color of divider for completed steps        |
+| `KStepDividerColorDefault`    | Color of divider for non-completed steps    |
+| `KStepCompletedColor`         | Primary color of completed state icon       |
+| `KStepCompletedSecondaryColor | Secondary color of completed state icon     |
```
Comment (adamdehaven) [inline docs/components/stepper.md:124]:
```suggestion
| `KStepCompletedSecondaryColor` | Secondary color of completed state icon     |
```

File: docs/components/stepper.md [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+<KStepper :steps="longSteps" max-label-width="100" />
+
+```html
+<KStepper :steps="steps" max-label-width="100" />
+```
+## Theming
+
+| Variable                      | Purpose                                     |
+| :---------------------        | :-------------------------------            |
+| `KStepSize`                   | Height and width of step state icon         |
```
Comment (adamdehaven) [inline docs/components/stepper.md:119]: Let's just remove this one and hard-code the size (all instances)

File: docs/components/stepper.md (hunk 5 above)
Comment (adamdehaven) [inline docs/components/stepper.md:124]:
```suggestion
| `KStepCompletedSecondaryColor` | Secondary color of completed state icon     |
```

File: docs/components/stepper.md (hunk 6 above)
Comment (kaiarrowood) [reply docs/components/stepper.md]: Fixed

File: docs/components/stepper.md (hunk 5 above)
Comment (kaiarrowood) [reply docs/components/stepper.md]: Fixed

File: docs/components/stepper.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+### maxLabelWidth
+
+The width of step labels (default is `170px`). We support numbers (will be converted to `px`), `auto`, and percentages (e.g. `25%`) for values.
+
+<KStepper :steps="longSteps" max-label-width="100" />
+
+```html
+<KStepper :steps="steps" max-label-width="100" />
+```
+## Theming
```
Comment (adamdehaven) [inline docs/components/stepper.md:115]: nit: need an empty line here for formatting
```suggestion

## Theming
```

File: src/components/KStepper/KStep.vue [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+        <KLabel>
+          {{ label }}
+        </KLabel>
+      </div>
+    </div>
+  </li>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed } from 'vue'
```
Comment (adamdehaven) [inline src/components/KStepper/KStep.vue:26]:
```suggestion
import { defineComponent, computed, PropType } from 'vue'
```

File: src/components/KStepper/KStep.vue [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+export default defineComponent({
+  name: 'KStep',
+  components: { KLabel, KStepState },
+  props: {
+    label: {
+      type: String,
+      required: true,
+    },
+    state: {
+      type: String,
```
Comment (adamdehaven) [inline src/components/KStepper/KStep.vue:42]:
```suggestion
      type: String as PropType<StepperState>,
```

File: src/components/KStepper/KStep.vue [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+      </div>
+    </div>
+  </li>
+</template>
+
+<script lang="ts">
+import { defineComponent, computed } from 'vue'
+import useUtilities from '@/composables/useUtilities'
+import KLabel from '@/components/KLabel/KLabel.vue'
+import KStepState from '@/components/KStepper/KStepState.vue'
```
Comment (adamdehaven) [inline src/components/KStepper/KStep.vue:29]:
```suggestion
import KStepState, { StepperState } from '@/components/KStepper/KStepState.vue'
```

File: src/components/KStepper/KStepState.vue [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
+import KCompletedState from '@/components/KStepper/stepper-icons/KCompletedState.vue'
+import KDefaultState from '@/components/KStepper/stepper-icons/KDefaultState.vue'
+import KErrorState from '@/components/KStepper/stepper-icons/KErrorState.vue'
+import KPendingState from '@/components/KStepper/stepper-icons/KPendingState.vue'
+
```
Comment (adamdehaven) [inline src/components/KStepper/KStepState.vue:16]:
```suggestion

export type StepperState = '' | 'default' | 'pending' | 'completed' | 'error'

```

File: src/components/KStepper/KStepState.vue [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+export default defineComponent({
+  name: 'KStepState',
+  components: { KCompletedState, KDefaultState, KErrorState, KPendingState },
+  props: {
+    state: {
+      type: String,
```
Comment (adamdehaven) [inline src/components/KStepper/KStepState.vue:22]:
```suggestion
      type: String as PropType<StepperState>,
```

File: src/components/KStepper/KStepState.vue [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+  <div class="k-step-state px-3">
+    <KCompletedState v-if="state === 'completed'" />
+    <KErrorState v-else-if="state === 'error'" />
+    <KPendingState v-else-if="state === 'pending'" />
+    <KDefaultState v-else />
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue'
```
Comment (adamdehaven) [inline src/components/KStepper/KStepState.vue:11]:
```suggestion
import { defineComponent, PropType } from 'vue'
```

File: src/components/KStepper/KStepper.vue [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+      :state="step.state"
+      :max-label-width="maxLabelWidth"
+    />
+  </ol>
+</template>
+
+<script lang="ts">
+import { defineComponent, PropType } from 'vue'
+import KStep from '@/components/KStepper/KStep.vue'
+
```
Comment (adamdehaven) [inline src/components/KStepper/KStepper.vue:19]:
```suggestion
import { StepperState } from '@/components/KStepper/KStepState.vue'

```

File: src/components/KStepper/KStepper.vue [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+export interface StepItem {
+  label: string
+  state?: string
```
Comment (adamdehaven) [inline src/components/KStepper/KStepper.vue:22]:
```suggestion
  state?: StepperState
```

**Review by adamdehaven — COMMENTED** [review]
Added type interfaces for the `step` property; please verify on local that nothing is erroring

File: docs/components/stepper.md (hunk 7 above)
Comment (kaiarrowood) [reply docs/components/stepper.md:116]: Fixed

File: src/components/KStepper/KStepper.vue (hunk 14 above)
Comment (kaiarrowood) [reply src/components/KStepper/KStepper.vue:20]: Fixed

File: src/components/KStepper/KStepper.vue (hunk 15 above)
Comment (kaiarrowood) [reply src/components/KStepper/KStepper.vue]: Fixed

**Review by adamdehaven — APPROVED** [review]
lgtm

**Pull request merged**